omc3/plotting/utils/windows.py

```diff
@@ -0,0 +1,208 @@
+    )
+except ImportError as e:
+    LOG.debug(f"Could not import QtPy: {str(e)}")
+    QMainWindow, QApplication, QVBoxLayout  = None, None, None
+    QWidget, QTabWidget = object, object
+else:
+    from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+    from matplotlib.backends.backend_qt5agg import NavigationToolbar2QT as NavigationToolbar
+    try:
+        matplotlib.use('Qt5agg')
+    except ImportError as e:
+        if not "headless" in str(e):
+            raise
+        LOG.debug(
+            "Could not change mpl to use QT, "
+            "due to headless mode (i.e. no display connected)."
+            )
+
+# ------------------------------------------------------------------------------
+
+
+def is_qtpy_installed():
+    """Returns True if QtPy is installed."""
+    return QMainWindow is not None
+
+
+def log_no_qtpy_many_windows():
+    """Logs a warning that QtPy is not installed and many windows will be opened."""
+
+    LOG.warning(
+        "QtPy is not installed. "
+        "Plots will be shown in individual windows. "
+        "Install QtPy for a more organized representation. "
+    )
+
+
+class PlotWidget(QWidget):
+
+    def __init__(self, *figures: Figure, title: str):
+        """A derived QWidget that contains Matplotlib-Figures,
+        stacked vertically, each with its own toolbar.
+
+        Args:
+            *figures (Figure): Figures to be contained in the widget
+            title (str): Name of the widget
+            skip_super (bool): If True the superclass __init__ will be skipped (ONLY FOR TESTS)
+        """
+        QWidget.__init__(self)   # no super(), because will not work with tests
+
+        self.title: str = title
+        self.figures: Tuple[Figure] = figures
+        self._canvas_toolbar_handles: List[Tuple[FigureCanvas, NavigationToolbar]] = []
+
+        self._layout = QVBoxLayout()
+        self.setLayout(self._layout)
+
+        for figure in figures:
+            canvas = FigureCanvas(figure)
+            toolbar = NavigationToolbar(canvas, self)
+
+            self._layout.addWidget(canvas)
+            self._layout.addWidget(toolbar)
+            self._canvas_toolbar_handles.append((canvas, toolbar))
+
+
+class TabWidget(QTabWidget):
+
+    def __init__(self, title: str = None):
+        """A simple tab widget, that in addition
+        to QTabWidget keeps track of the tabs via dictionary
+        and can have a title.
+        
+        Args:
+            title (str): Title of the created Window
+            skip_super (bool): If True the superclass __init__ will be skipped (ONLY FOR TESTS)
+        """
+        QTabWidget.__init__(self)  # no super(), because will not work with tests
+
+        self.title: str = title
+        self.tabs: Dict[str, QWidget] = {}
+
+    def add_tab(self, widget: QWidget):
+        self.tabs[widget.title] = widget
+        self.addTab(widget, widget.title)
+
+
+class SimpleTabWindow:
+
```

Comment:
This is a full-blown windows application that runs its event loop, right? Maybe let's give it a short docstring to specify that. It would help for Qt-noobs like me

Reply:
the docstring is in the init. but I added a sentence